### PR TITLE
fix: fix primary key for allocation_accelerators table [DET-9890]

### PR DESCRIPTION
**Pull request by salonig23**
## Description
slurm tests were failing because container id isn't unique in slurm jobs. Removed primary key constraint on container_id and added a new primary key

## Test Plan

- tested manually to see if migrations run properly. 
- slurm e2e tests should pass

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-9890

### CODE REVIEW

File: master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE public.allocation_accelerators
+    DROP CONSTRAINT allocation_accelerators_pkey,
+    ADD COLUMN allocation_accelerators_id SERIAL,
```
Comment (NicholasBlaskey) [inline master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql:3]: I think we can just call this column ```id```

Comment (NicholasBlaskey) [inline master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql:3]: nit: don't use serial https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_serial

instead do ```INT GENERATED BY DEFAULT AS IDENTITY```

File: master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE public.allocation_accelerators
+    DROP CONSTRAINT allocation_accelerators_pkey,
```
Comment (NicholasBlaskey) [inline master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql:2]: I think you need to update the bun model with ```ID            int          `bun:"id,pk,autoincrement"````

File: master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql (hunk 1 above)
Comment (salonig23) [reply master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql]: Done!

File: master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql (hunk 2 above)
Comment (salonig23) [reply master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql:2]: Done!

File: master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql (hunk 1 above)
Comment (salonig23) [reply master/static/migrations/20231010095640_fix-accelerator-data-table.tx.up.sql]: done!

**Review by NicholasBlaskey — APPROVED** [review]
LGTM

**Pull request merged**